includes/models/model.llms.quiz.php

```diff
@@ -213,7 +226,72 @@ public function can_be_resumed_by_student( $user_id = null ) {
 			$can_be_resumed_by_student = $last_attempt && $last_attempt->can_be_resumed();
 		}
 
-		return $can_be_resumed;
+		// Check if the quiz can be resumed by a student and if the resume time limit is not reached.
+		if ( $can_be_resumed_by_student && $this->has_resume_attempt_time_expired() ) {
+			$can_be_resumed_by_student = false;
+		}
+
+		return $can_be_resumed_by_student;
+
+	}
+
+	/**
+	 * Determine if the student's resume quiz attempt time limit is expired.
+	 *
+	 * The resume quiz attempt time limit for a student is 1 day.
+	 *
+	 * @since [version]
+	 *
+	 * @return bool
+	 */
+	private function has_resume_attempt_time_expired() {
+
+		$student = llms_get_student();
+
+		if ( ! $student ) {
+			return false;
+		}
+
+		$last_attempt = $student->quizzes()->get_last_attempt( $this->get( 'id' ) );
+
+		if ( ! $last_attempt->get( 'start_date' ) ) {
+			return false;
+		}
+
+		$start_date = $last_attempt->get( 'start_date' );
+
+		/**
+		 * Filters the X time for resuming quiz.
+		 *
+		 * @since [version]
+		 *
+		 * @param int $time_period The time period in days.
+		 */
+		$time_period     = apply_filters( 'llms_quiz_resume_time_period', 1 );
+		$expiration_date = strtotime( "+{$time_period} days", strtotime( $start_date ) );
+		$current_date    = llms_current_time( 'timestamp' );
+
+		return $current_date > $expiration_date;
+	}
+
+	/**
+	 * Gets quiz's last attempt key of a user.
+	 *
+	 * @since [version]
+	 *
+	 * @param int $user_id Optional. WP User ID, none supplied uses current user. Default `null`.
+	 * @return bool
+	 */
+	public function get_student_last_attempt_key( $user_id = null ) {
+
+		$student      = llms_get_student( $user_id );
+		$last_attempt = false;
+
+		if ( $student ) {
+			$last_attempt = $student->quizzes()->get_last_attempt( $this->get( 'id' ) )->get_key();
```

Comment:
`$student->quizzes()->get_last_attempt( $this->get( 'id' ) )`
this can return `false`, check it's not before trying to get its key.

Reply:
What do you mean with "it's working fine"?
Again:
`$student->quizzes()->get_last_attempt( $this->get( 'id' ) );`
can return false if there area no attempts made by the student for the quiz.

`get_student_last_attempt_key` is a public method, meaning that potentially anybody can call it.
If you call this method for a student who never did an attempt you'll get:
`Uncaught Error: Call to a member function get_key() on bool` kind of error.